modules/ossm-extensions-ref-wasmplugin.adoc

```diff
@@ -0,0 +1,175 @@
+////
+This module included in the following assemblies:
+*service_mesh_/v2x/ossm-extensions.adoc
+////
+:_content-type: REFERENCE
+[id="ossm-wasm-ref-wasmplugin_{context}"]
+= WasmPlugin API reference
+
+The WasmPlugins API provides a mechanism to extend the functionality provided by the Istio proxy through WebAssembly filters.
+
+You can deploy multiple WasmPlugins. The order of execution (as part of Envoy’s filter chain) is determined by the `phase` and `priority` settings, allowing the configuration of complex interactions between user-supplied WasmPlugins and Istio’s internal filters.
+
+In the following example, an authentication filter is deployed to the ingress-gateway that implements an OpenID flow and populates the `Authorization` header with a JSON Web Token (JWT) to be consumed by Istio authentication.
```

Comment:
I found this sentence a bit hard to follow. Inverting it a little bit makes it easier for me. Something like:

In the following example, an authentication filter that implements an OpenID flow and populates the `Authorization` header with a JSON Web Token (JWT) to be consumed by Istio authentication is deployed to the ingress gateway. The WasmPlugin file lives in the proxy sidecar filesystem. Note the field `url`.